projects/optic/src/config.ts

```diff
@@ -58,18 +65,19 @@ const CaptureConfigData = Type.Object({
     ready_interval: Type.Optional(Type.Number()),
     ready_timeout: Type.Optional(Type.Number()),
   }),
-  requests: Type.Optional(Type.Array(Request)),
-  requests_command: Type.Optional(
+  requests: Type.Optional(
     Type.Object({
-      command: Type.String(),
-      proxy_variable: Type.Optional(Type.String()),
+      // one of these is technically required, but that's enforced at runtime
+      run: Type.Optional(RequestRun),
+      send: Type.Optional(Type.Array(RequestSend)),
```

Comment:
@niclim is there a good way to express that at least one of these needs to exist at the type level? would avoid needing to make a runtime check.

Reply:
I think you can do a `OneOf` union of 
```
Type.OneOf([ // not sure the exact signature of this
Type.Object({  run: RequestRun, send: Type.Optional(Type.Array(RequestSend)), }),
Type.Object({  run: Type.Optional(RequestRun), send: Type.Array(RequestSend), })
])
```

I'm not really sure if that's clearer - what you have is fine though and will work

Reply:
ill take a quick look. i think in the long run it is clearer to do it here, otherwise we need to remember that there's random validation logic in the actual capture flow. i like the idea of knowing before we get there whether or not the config is 100% legit.

Reply:
after looking i don't think typebox does oneOf.